### PR TITLE
Smoothing size

### PR DESCRIPTION
**Pull request by caspervdw**
This tackles #401 for the v0.4 release

### CODE REVIEW

**Comment by nkeim** [comment]
The changes to the code look good! It looks like some of the feature-finding tests need to be updated to match the new API. Is that true?

**Comment by danielballan** [comment]
Any reason not to keep `minmass_version_change` around for users who pick up some old work from v0.2.x and migrate it straight to v0.4.x?

**Comment by caspervdw** [comment]
@danielballan  You're right, we should keep the `minmass_version_change`. I now figured out what changed in this PR so that the original tests broke. Because of a reduced (and square! see #361) boxcar kernel, features are dimmer after the bandpassing, leading to lower `mass` values.

This makes sense: a rolling average with a smaller kernel is less smoothed out than one with a larger kernel.  If you subtract this from the original image, the smaller the kernel, the dimmer the features. I explained this in the docstring of the new `minmass_v04_change`.

Counterintuitively, the locate precision increased in this PR: `ep` is now always larger than the real precision (see last commit to `test_ep`). One reason for this is described in #401. I now added an explicit test for this. `ep` can be seen as an upper boundary to the precision.

**Comment by caspervdw** [comment]
Tests pass! This turned out to be a significant change in the locate outcome. See the list in my previous post, which I updated.

File: trackpy/feature.py [hunk 1]
```diff
@@ -73,7 +73,84 @@ def minmass_version_change(raw_image, old_minmass, preprocess=True,
 
     scale_factor = scalefactor_to_gamut(image, dtype)
 
-    return int(old_minmass / scale_factor)
+    return old_minmass / scale_factor
+
+
+def minmass_v04_change(raw_image, old_minmass, diameter, preprocess=True,
+                       old_smoothing_size=None, new_smoothing_size=None):
+    """Convert minmass value from v0.3 o v0.4.
```
Comment (nkeim) [inline trackpy/feature.py]: v0.3 **to** v0.4

File: trackpy/feature.py [hunk 2]
```diff
@@ -73,7 +73,84 @@ def minmass_version_change(raw_image, old_minmass, preprocess=True,
...
+
+    From trackpy version 0.4.0, the default image preprocessing is changed.
+    Before version 0.4.0 a blurred out image (rolling or boxcar average) with
+    a circular kernel with radius `diameter` was subtracted from the image
+    before refinement and mass calculation. From version 0.4.0, this has
+    changed to a square kernel with sides `diameter`, more or less twice as
+    small. This increasing tracking accuracy, and reduces the mass.
```
Comment (nkeim) [inline trackpy/feature.py]: increases

File: trackpy/preprocessing.py [hunk 3]
```diff
@@ -116,18 +118,23 @@ def bandpass(image, lshort, llong, threshold=None, truncate=4):
     See Also
     --------
     lowpass, boxcar, legacy_bandpass, legacy_bandpass_fftw
+
+    Notes
+    -----
+    The boxcar size changed in v0.4: before, one side of the square kernel
```
Comment (nkeim) [inline trackpy/preprocessing.py]: Maybe note that the shape also changed?

**Comment by caspervdw** [comment]
@nkeim Done and ready to merge.

**Comment by caspervdw** [comment]
This seems to me a Travis hiccup. Let's see what it does tomorrow.

**Pull request merged**